app/services/reports/region_cache_warmer.rb

```diff
@@ -24,36 +24,15 @@ def call
           return
         end
 
-        notify "starting state caching"
-        Statsd.instance.time("region_cache_warmer.states") do
-          cache_states
-        end
-        Statsd.instance.time("region_cache_warmer.with_exclusions.states") do
-          cache_states(with_exclusions: true)
-        end
-
-        notify "starting facility_group caching"
-        Statsd.instance.time("region_cache_warmer.facility_groups") do
-          cache_facility_groups
-        end
-        Statsd.instance.time("region_cache_warmer.with_exclusions.facility_groups") do
-          cache_facility_groups(with_exclusions: true)
-        end
+        notify "starting region caching"
+        Statsd.instance.time("region_cache_warmer") do
+          Region.where.not(region_type: ["root", "organization"]).find_each(batch_size: BATCH_SIZE) do |region|
+            RegionService.call(region: region, period: period)
+            Statsd.instance.increment("region_cache_warmer.#{region.region_type}.cache")
 
-        notify "starting block caching"
-        Statsd.instance.time("region_cache_warmer.blocks") do
-          cache_blocks
-        end
-        Statsd.instance.time("region_cache_warmer.with_exclusions.blocks") do
-          cache_blocks(with_exclusions: true)
-        end
-
-        notify "starting facility caching"
-        Statsd.instance.time("region_cache_warmer.facilities") do
-          cache_facilities
-        end
-        Statsd.instance.time("region_cache_warmer.with_exclusions.facilities") do
-          cache_facilities(with_exclusions: true)
+            RegionService.call(region: region, period: period, with_exclusions: true)
+            Statsd.instance.increment("region_cache_warmer.with_exclusions.#{region.region_type}.cache")
+          end
```

Comment:
Nice. We'll still get individual stats on facility cache times vs state cache times, etc, right?